xarray/tests/test_units.py

```diff
@@ -78,17 +167,19 @@ def array_extract_units(obj):
     if isinstance(obj, (xr.Variable, xr.DataArray, xr.Dataset)):
         obj = obj.data
 
-    try:
-        return obj.units
-    except AttributeError:
-        return None
+    for unit_lib in unit_libs:
+        if isinstance(obj, unit_lib.quantity_type):
+            return unit_lib.get_unit(obj)
+
+    return None
 
 
 def array_strip_units(array):
-    try:
-        return array.magnitude
-    except AttributeError:
-        return array
+    for unit_lib in unit_libs:
+        if isinstance(array, unit_lib.quantity_type):
+            return unit_lib.strip_units(array)
+
+    return array
```

Comment:
I wonder if it would make sense to define a function that determines the unit info instance (or have a dict of quantity type → unit info and use that to look up the unit info)? That way we don't have these loops everywhere.

```suggestion
    unit_lib = units_libs.get(type(array), None)
    if unit_lib is not None:
        return unit_lib.strip_units(array)
    return array
```

Reply:
👍